src/Microsoft.SqlTools.ServiceLayer/Metadata/SmoScripterFactory.cs

```diff
@@ -0,0 +1,191 @@
+﻿//
+// Copyright (c) Microsoft. All rights reserved.
+// Licensed under the MIT license. See LICENSE file in the project root for full license information.
+//
+
+using System.Collections.Generic;
+using System.Data.Common;
+using System.Threading.Tasks;
+using Microsoft.Data.SqlClient;
+using Microsoft.SqlServer.Management.Common;
+using Microsoft.SqlServer.Management.Smo;
+using Microsoft.SqlTools.SqlCore.Connection;
+using Microsoft.SqlTools.ServiceLayer.Connection.ReliableConnection;
+
+
+namespace Microsoft.SqlTools.ServiceLayer.Metadata
+{
+    internal static class SmoScripterFactory
```

Comment:
nit: But I don't really see how this is a "Scripter Factory" at all (it's not making a script). Something like `SmoScripterHelpers` would be better IMO